providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -210,197 +229,58 @@ public String getName() {
     @Override
     public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue,
             EvaluationContext ctx) {
-        
-        if (this.cacheAvailable()) {
-            ProviderEvaluation<Value> fromCache = this.cache.get(key);
-            if (fromCache != null) {
-                ProviderEvaluation<Boolean> result = ProviderEvaluation.<Boolean>builder()
-                    .value(fromCache.getValue().asBoolean())
-                    .variant(fromCache.getVariant())
-                    .reason(CACHED_REASON)
-                    .build();
-                return result;
-            }
-        }
-
-        ResolveBooleanRequest request = ResolveBooleanRequest.newBuilder()
-                .setFlagKey(key)
-                .setContext(this.convertContext(ctx))
-                .build();
-        ResolveBooleanResponse r = this.serviceBlockingStub.withDeadlineAfter(this.deadline, TimeUnit.MILLISECONDS)
-                .resolveBoolean(request);
-        ProviderEvaluation<Boolean> result = ProviderEvaluation.<Boolean>builder()
-                .value(r.getValue())
-                .variant(r.getVariant())
-                .reason(r.getReason())
-                .build();
 
+        ResolveBooleanRequest request = ResolveBooleanRequest.newBuilder().buildPartial();
 
-        if (this.isEvaluationCacheable(result)) {
-            ProviderEvaluation<Value> value = ProviderEvaluation.<Value>builder()
-                .value(new Value(r.getValue()))
-                .variant(r.getVariant())
-                .reason(r.getReason())
-                .build();
-            this.cache.put(key, value);
-        }
-        
-        return result;
+        return this.resolve(key, ctx, request,
+                this.serviceBlockingStub.withDeadlineAfter(this.deadline, TimeUnit.MILLISECONDS)::resolveBoolean, null);
```

Comment:
This is the TLDR of this change... now we call `this.resolve` with a `request` and a resolver function ref (`::resolveBoolean`) instead of duplicating this stuff.

Reply:
Nice! Making a note to do something similar in the Go provider